### ISSUE TITLE
Tutorial request

### ISSUE THREAD

**Issue by apoorvmote** [nofelmahmood/CKSIncrementalStore#2]
There is too much going on in the app. 

I tried to recreate the app from scratch but its not working. I think I have missed to add something. 

If you could post tutorial on medium then it would be great.

Thanks either way for this project.

**Comment by nofelmahmood** [comment]
No Problem. I am glad you liked it.
Kindly check out the demo project again and try to work out something from there.
btw Still I'l consider simplifying the read me of the repository. :)

**Comment by apoorvmote** [comment]
Also because I only know swift & never learned objective C. So I am bit confused. 

And learning from your code is my top priority at the moment. I am spending 6-8 hours per day reading your code & Apple documentation.

If you create tutorial then I will learn faster. But one way or the other I will reverse engineer your code.

**Comment by nofelmahmood** [comment]
I appreciate your enthusiasm. Currently I had plans to write how I implemented the sync part. I started it and I think I might complete it in a day or two. Best of luck in learning swift but my advice for you is Think Big, Start Small. First consider doing some small stuff in swift and then build up from there.
I'l post the link to the article (Related to sync) on this issue when I am finished till then best of luck :)

**Comment by apoorvmote** [comment]
Yeah its frustrating. I studied core data from e book & cloud kit from course I purchased. But they don't teach you how to combine them together. 

I was like why the hell did I buy this course. The only reason I bought it so I can use them together in app I am building. 

I have been studying swift for about 8 months & core data / cloud kit for last 3 months. Hopefully I will learn this quicker. 

Also please don't forget to post link here when you create tutorial.

**Comment by apoorvmote** [comment]
Your app actually has one issue. When you delete item the app crashes.
When you delete one row you didn't delete from tasks array.
You just need to add following line

self.tasks.removeAtIndex(indexPath.row)

**Comment by apoorvmote** [comment]
I have found a code that would tell you whether user changed their iCloud account or logged out of iCloud in middle of using the app.

```
func applicationDidBecameActive(notification:NSNotification) {

    println("Application is active")

    NSNotificationCenter.defaultCenter().removeObserver(self, name: NSUbiquityIdentityDidChangeNotification, object: nil)

}

func applicationBecameInactive(notification:NSNotification) {

    println("Application is inactive")

    NSNotificationCenter.defaultCenter().addObserver(self, selector: "handleIdentityChange:", name: NSUbiquityIdentityDidChangeNotification, object: nil)

}

func handleIdentityChange(notification:NSNotification) {

    println("this is working")

    let fileManager = NSFileManager()

    if let token = fileManager.ubiquityIdentityToken {

        println("New token is \(token)")

    }
    else {

        println("User has logged out of iCloud")

    }

}
```

Waiting for your sync tutorial. Please create it soon. Its very difficult to understand otherwise.
